controllers/contact-controller.js

```diff
@@ -67,6 +67,80 @@ function isEmptySearchParams(body) {
 
 const orUndefined = value => value || undefined;
 
+const queryOnName = operator => singleInput => firstName => lastName =>
+  (singleInput || firstName || lastName) && {
+    [Op.or]: [
+      {
+        [Op.and]: [
+          {
+            'rawJson.callType': {
+              [Op.in]: [callTypes.child, callTypes.caller],
+            },
+          },
+          {
+            [operator]: [
+              (firstName || singleInput) && {
+                'rawJson.childInformation.name.firstName': {
+                  [Op.iLike]: `%${singleInput || firstName}%`,
+                },
+              },
+              (lastName || singleInput) && {
+                'rawJson.childInformation.name.lastName': {
+                  [Op.iLike]: `%${singleInput || lastName}%`,
+                },
+              },
+            ],
+          },
+        ],
+      },
+      {
+        [Op.and]: [
+          {
+            'rawJson.callType': callTypes.caller,
+          },
+          {
+            [operator]: [
+              (firstName || singleInput) && {
+                'rawJson.callerInformation.name.firstName': {
+                  [Op.iLike]: `%${singleInput || firstName}%`,
+                },
+              },
+              (lastName || singleInput) && {
+                'rawJson.callerInformation.name.lastName': {
+                  [Op.iLike]: `%${singleInput || lastName}%`,
+                },
+              },
+            ],
+          },
+        ],
+      },
+    ],
+  };
+
+const queryOnPhone = singleInput => phoneNumber => {
+  const re = /[\D]/gi;
+  const singleDigitsOnly = singleInput && singleInput.replace(re, '');
+  const phoneDigitsOnly = phoneNumber && phoneNumber.replace(re, '');
+
+  // column should be passed via Sequelize.col or Sequelize.literal
+  const phoneRegExp = column =>
+    Sequelize.where(Sequelize.fn('REGEXP_REPLACE', column, '[^[:digit:]]', '', 'g'), {
+      [Op.iLike]: `%${singleDigitsOnly || phoneDigitsOnly}%`,
```

Comment:
This will not find a number in the following situation:
- The counselor types a number into the form like `(212) 555-1212`.  Notice that the country code is omitted.
- Later, someone searches for `+12125551212`, which includes the country code.  This is especially likely if it's an automated search.

That would yield an evaluation of `WHERE '2125551212' iLike '%12125551212%'`, which would be `false` because the country code "1" is not present.  So we would not return that record.

I think I'm okay with that for now.  But could you please add a unit test that explicitly tests that example and shows that it fails, so we have a form of documentation of this?